### PR TITLE
Add error to RHS reply box for messages > 4000 chars, consistent with create post and edit post errors

### PR DESCRIPTION
**Pull request by fraziern**
#### Summary
Adds warning when a reply is over 4000 chars, and flashes warning when attempting to submit. UX is intended to be consistent with PRs #4789 and #4687.

#### Ticket Link
#4856 
[Jira Ticket](https://mattermost.atlassian.net/browse/PLT-4976)

#### Checklist
- [x] Has UI changes

### CODE REVIEW

**Comment by yangchen1** [comment]
+1. Thanks for the fix @fraziern!

File: webapp/components/create_comment.jsx [hunk 1]
```diff
@@ -263,6 +263,25 @@ export default class CreateComment extends React.Component {
         $('.post-right__scroll').parent().scrollTop($('.post-right__scroll')[0].scrollHeight);
 
         this.setState({message});
+
+        this.checkMessageLength(message);
+    }
+
+    checkMessageLength(message) {
```
Comment (enahum) [inline webapp/components/create_comment.jsx]: Wondering if we should have this function be a common function to use here and https://github.com/mattermost/platform/pull/4867/files#diff-9c47c7c15d2564d7af567a75ba50ce9dR228

**Comment by fraziern** [comment]
@enahum good suggestion. Looks like the `Textbox` component is a dependency only in `CreateComment`, `CreatePost`, and `EditPostModal`, so maybe moving it there would work better. I can give it a try.

File: webapp/components/create_comment.jsx (hunk 1 above)
Comment (fraziern) [inline webapp/components/create_comment.jsx]: great suggestion @enahum, let me take a look at how we might be able to accomplish that. Maybe by moving most of the code to the shared `Textbox` component...

**Comment by fraziern** [comment]
@enahum yeah, broadened the scope a bit to refactor the prior PRs and make them all more cohesive. So in addition to adding the error to the RHS:
 
 1. The message length check is moved to the `Textbox` component, which all 3 message editing/drafting components use
 2. Added a `handlePostError` callback to the 3 drafting components, to let `Textbox` update their state the same way the message length check did previously

Let me know what you think.

**Pull request merged**